### PR TITLE
Blog Post Management (CRUD)

### PR DESCRIPTION
**Pull request by muhammadjon8**

### CODE REVIEW

File: src/entities/entity.blog.ts
Comment (mukhammadjon-ss) [inline src/entities/entity.blog.ts:1]: naming should be proper. 
application is blog application entity is article or post. article.entity.ts, post.entity.ts

File: src/entities/entity.user.ts
Comment (mukhammadjon-ss) [inline src/entities/entity.user.ts:1]: user.entity.ts

File: src/express.d.ts
Comment (mukhammadjon-ss) [inline src/express.d.ts:1]: I think you don't need this namespace. User can be taken from token. in prev task, you implemented auth so user can be retrieved from toke. request should not include all user info

File: src/routes/blogRoutes.ts
Comment (mukhammadjon-ss) [inline src/routes/blogRoutes.ts:1]: why routes are not protected ?
create update, delete must be proctected

File: src/controllers/controller.blog.ts
Comment (mukhammadjon-ss) [inline src/controllers/controller.blog.ts:1]: blog.contoller.ts

File: src/services/blogs.services.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      where: { id: blogId },
+      relations: ['author'],
+    })
+
+    if (!blog) {
+      throw new Error('Blog not found')
+    }
+    const user = await this.userRepo.findOne({where: { id: author }})
+
+    if (blog.author.id !== author && user?.isAdmin == false) {
```
Comment (mukhammadjon-ss) [inline src/services/blogs.services.ts]: this can separate middleware

File: src/controllers/controller.blog.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  try {
+    const blog = BlogService.createBlog(userId, title, content)
+    res.status(201).json({ message: 'Blog created successfully', data: blog })
+  } catch (error) {
+    res.status(400).json({ message: error })
+  }
+}
+
+export const updateBlogById = (req: Request, res: Response) => {
+  const { title, content, author } = req.body
```
Comment (mukhammadjon-ss) [inline src/controllers/controller.blog.ts]: you have to take user from token. not from request body.

**Review by mukhammadjon-ss — CHANGES_REQUESTED** [review]
be more consistent in naming

**Comment by muhammadjon8** [comment]
Yeah, I will change them

File: src/controllers/blog.controller.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import BlogService from '../services/blogs.services'
+import { BlogsSearchParamsDto } from '../dtos/blog.searchparams.dto'
+
+export const createBlog = async (req: any, res: any) => {
+  console.log('Request Body:', req.body.title)
```
Comment (mukhammadjon-ss) [inline src/controllers/blog.controller.ts]: remove console.log

File: src/controllers/blog.controller.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  const { title, content, author } = req.body
+
+  console.log(content, title, author)
```
Comment (mukhammadjon-ss) [inline src/controllers/blog.controller.ts]: remove console.log

**Review by mukhammadjon-ss — CHANGES_REQUESTED** [review]

File: src/services/blogs.services.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    const user = await this.userRepo.findOne({ where: { id: author } })
+
+    if (blog.author.id !== author && user?.isAdmin == false) {
+      throw new Error('Unauthorized')
+    }
+
+    blog.title = title
+    blog.content = content
+    const isAuthor = await this.userRepo.findOne({ where: { id: author } })
+    if (isAuthor) {
```
Comment (mukhammadjon-ss) [inline src/services/blogs.services.ts:37]: this is redundant. This is protected route and it has already auth middleware which checks and authenticates user.

**Review by mukhammadjon-ss — CHANGES_REQUESTED** [review]

**Comment by muhammadjon8** [comment]
Fixed!

пн, 14 окт. 2024 г. в 18:36, Mukhammadjon Sirojiddinov <
***@***.***>:

**Pull request merged**